Source/Core/DolphinQt/DiscordJoinRequestDialog.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#ifdef USE_DISCORD_PRESENCE
+
+#include "DiscordJoinRequestDialog.h"
+
+#include <QGridLayout>
+#include <QLabel>
+#include <QPixmap>
+#include <QPushButton>
+#include <discord-rpc/include/discord_rpc.h>
+#include "Common/HttpRequest.h"
+
+DiscordJoinRequestDialog::DiscordJoinRequestDialog(QWidget* parent, const char id[],
+                                                   const std::string& discord_tag,
+
+                                                   const char avatar[])
+    : QDialog(parent), m_user_id(id), m_close_timestamp(std::time(nullptr) + m_max_lifetime)
+{
+  setWindowTitle(tr("Request to Join Your Party"));
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+  QPixmap avatar_pixmap;
+
+  if (avatar[0] != '\0')
+  {
+    const std::string path = "https://cdn.discordapp.com/avatars/";
+    const std::string file_type = ".png";
+    std::string avatar_endpoint;
+    avatar_endpoint.reserve(path.length() + strlen(id) + 1 + strlen(avatar) + file_type.length());
+    avatar_endpoint += path;
+    avatar_endpoint += id;
+    avatar_endpoint += '/';
+    avatar_endpoint += avatar;
+    avatar_endpoint += file_type;
+
+    Common::HttpRequest request;
+    Common::HttpRequest::Response response = request.Get(avatar_endpoint);
+
+    if (response.has_value())
+      avatar_pixmap.loadFromData(response->data(), static_cast<uint>(response->size()), "png");
+  }
+
+  CreateMainLayout(discord_tag, avatar_pixmap);
+  ConnectWidgets();
+}
+
+const std::time_t DiscordJoinRequestDialog::GetCloseTimestamp()
```

Comment:
This const here isn't necessary. This function can be a const member function, however, like:

```cpp
std::time_t DiscordJoinRequestDialog::GetCloseTimestamp() const
```